PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/ConnectionManager.java

```diff
@@ -84,10 +86,15 @@ private static String typeOf(String resourceType, String consumerType) {
 
         @Override
         public synchronized void addConnection(Connection<? extends Resource, ? extends Resource> connection) {
-            connections.remove(connection); // always replace the old with the new one.
+            connections.removeIf(c -> Objects.equals(c, connection)); // always replace the old with the new one.
```

Comment:
as far as I know, `remove(connection)` equals `removeIf(c -> Objects.equals(c, connection))`

Reply:
yes. the cost is the complexity (the "n" is small and acceptable currently), I thought uniform "removeIf" method in this class is also important. and we shouldn't depend on "synchronized" on the method to prevent ConcurrentModificationException.